framework/src/outputs/png/PNGOutput.C

```diff
@@ -25,8 +26,12 @@ validParams<PNGOutput>()
   params.addParam<bool>("transparent_background",
                         false,
                         "Determination of whether the background will be transparent.");
-  params.addParam<unsigned int>("resolution", 1, "The resolution of the image.");
-  MooseEnum color("GRAY BRYW BCR RWB BR");
+  params.addParam<unsigned int>("resolution", 25, "The resolution of the image.");
+  params.addParam<std::string>(
+      "variable", "None", "The name of the variable to use when creating the image");
+  params.addParam<Real>("max", -1000, "The maximum for the variable we want to use");
```

Comment:
We have a way of testing whether a parameter was set by the user. If you use that you don't have to attempt to find a value that is "out of range" or invalid. I'd set this to one and the min to zero.